timeutil/schedule.go

```diff
@@ -82,40 +108,285 @@ func ParseTime(s string) (t TimeOfDay, err error) {
 	return t, nil
 }
 
-// Schedule defines a start and end time in which events should run.
-type Schedule struct {
+// Weekday represents a weekday such as Monday, Tuesday, with optional
+// week-in-the-month position, eg. the first Monday of the month
+type Weekday struct {
+	// Day stands for a weekday - "mon", "tue", ..
+	Day string
```

Comment:
Good point. I used string here so that I could tell if it's unset and then such Weekday becomes a match-all. With your other suggestions, there will be no fallback, empty WeekSpans to match against, so switching to time.Weekday should be just fine.